pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go

```diff
@@ -120,11 +120,16 @@ func (e *AzureMonitorDatasource) buildQueries(queries []backend.DataQuery, dsInf
 			dimSB.WriteString(fmt.Sprintf("%s eq '%s'", dimension, dimensionFilter))
 		} else {
 			for i, filter := range azJSONModel.DimensionFilters {
-				dimSB.WriteString(filter.String())
-				if i != len(azJSONModel.DimensionFilters)-1 {
-					dimSB.WriteString(" and ")
+				if filter.Operator != "eq" && filter.Filter == "*" {
+					dimSB.WriteString(fmt.Sprintf("%s eq '*'", filter.Dimension))
+				} else {
+					dimSB.WriteString(filter.String())
+					if i != len(azJSONModel.DimensionFilters)-1 {
```

Comment:
should the code that adds `" and "` be outside this else?

Reply:
Great catch, missed this 😊 Explains why the below test was passing when it shouldn't of!